modules/querying/pages/query-optimizer/enable-cost-optimizer.adoc

```diff
@@ -0,0 +1,96 @@
+= Enabling Cost-Based Optimizer
+:sectnums:
+:description: Steps to enable the cost-based optimizer.
+
+This page walks you through the steps to enable the cost-based optimizer.
+
+== Before you begin
+* You have created a query that contains a xref:select-statement/index.adoc[`SELECT` statement] that uses a xref:select-statement/index.adoc#_path_pattern[path pattern] in its `FROM` clause with one or more hops.
+* You have the privilege to install queries on the graph where you are operating.
+
+== Procedure
+The following is the procedure to enable cost-based optimizer when installing a query.
+
+=== Identify referenced types and attributes
+You need to identify the vertex and edge types for which you need to compute cardinality data, and attributes of those types for which you need to compute histogram data.
+
+Vertex and edge types:
+
+* Any vertex type referenced in the path pattern of a `FROM` clause.
+* Any _refined edge type_.
+A _refined edge type_ means an edge type with a specified pair of `from` and `to` vertex types.
+
+Attributes:
+
+* Any attribute referenced in a xref:select-statement/index.adoc#_where[`WHERE` clause] in a `SELECT` statement in the query.
+
+For example, if you want to optimize the following query:
+
+[.wrap,gsql]
+----
+CREATE OR REPLACE QUERY example3() FOR GRAPH ldbc_snb SYNTAX V2 {
+
+ vSet = SELECT p
+        FROM Tag:t - (<HAS_TAG) - Comment:m - (<LIKES:e) - Person:p
+        WHERE p.gender == "female" AND t.id != 100;
+
+}
+----
+
+We would need the following statistics:
+
+* Cardinality
+** Vertex types
+*** `Tag`
+*** `Comment`
+*** `Person`
+** Refined edge types
+*** `HAS_TAG`: from `Comment` to `Tag`
+*** `LIKES`: from `Person` to `Comment`
```

Comment:
do you mean like "HAS_TAG_REVERSE",  suppose that's the name reverse edge is defined with?

since refined edge count will not be computed automatically for the reverse edge as well when computing HAS_TAG, we should.